src/argus/notificationprofile/models.py

```diff
@@ -278,14 +282,19 @@ def incident_fits(self, incident: Incident):
         if self.is_empty:
             return False  # Filter is empty!
         data = self.filter
-        source_fits = self.source_system_fits(incident, data)
-        tags_fit = self.tags_fit(incident, data)
-        new_filters_fit = self.filter_wrapper.incident_fits_tristates(
-            incident
-        ) and self.filter_wrapper.incident_fits_maxlevel(incident)
-        # If False then one filter failed
-        checks = set((source_fits, tags_fit, new_filters_fit))
-        return not (False in checks)  # At least one filter failed
+        checks = {}
+        checks["source"] = self.source_system_fits(incident, data)
+        checks["tags"] = self.tags_fit(incident, data)
+        checks["tristates"] = self.filter_wrapper.incident_fits_tristates(incident)
+        checks["max_level"] = self.filter_wrapper.incident_fits_maxlevel(incident)
+        # If False then at least one filter failed
+        final_result = True
+        for check, result in checks.items():
+            if result is False:
+                final_result = False
+                # Do not bail out early in order to log all failing checks
+                LOG.debug("Filter: incident failed %s check", check)
+        return final_result
```

Comment:
Here's an alternative suggestion from me:

```suggestion
        any_failed = False in checks.values()
        if any_failed:
           LOG.debug("Filter: at least one incident check failed: %r", checks)
           return False
        else:
           return True
```

My thinking here is that if any of the checks failed, we might as well log the whole `checks` dictionary, so all the results can be debugged.

Reply:
Or even just:
```suggestion
        any_failed = False in checks.values()
        if any_failed:
           LOG.debug("Filter: at least one incident check failed: %r", checks)
        return not any_failed
```